services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/SwapEndpointSpec.groovy

```diff
@@ -0,0 +1,481 @@
+                new SwapFlowPayload(flow1.id, flow1Src, flow1Dst),
+                new SwapFlowPayload(flow2.id, flow2Src, flow2Dst))
+
+        then: "Endpoints are successfully swapped"
+        verifyEndpoints(response, flow1Src, flow1Dst, flow2Src, flow2Dst)
+        verifyEndpoints(flow1, flow2, flow1Src, flow1Dst, flow2Src, flow2Dst)
+
+        and: "Flows validation doesn't show any discrepancies"
+        validateFlows(flow1, flow2)
+
+        and: "Switch validation doesn't show any missing/excess rules and meters"
+        validateSwitches(flow1SwitchPair)
+        validateSwitches(flow2SwitchPair)
+
+        and: "Delete flows"
+        [flow1, flow2].each { flowHelper.deleteFlow(it.id) }
+
+        where:
+        description << ["src <-> src", "dst <-> dst", "src <-> dst", "dst <-> src"]
+        flow1SwitchPair << [getTopologyHelper().getNotNeighboringSwitchPair()] * 4
+        flow2SwitchPair << [getDifferentNotNeighboringSwitchPair(flow1SwitchPair)] * 4
+        flow1 << [getFlowHelper().randomFlow(flow1SwitchPair)] * 4
+        flow2 << [getFlowHelper().randomFlow(flow2SwitchPair)] * 4
+        [flow1Src, flow1Dst, flow2Src, flow2Dst] << [
+                [flow2.source, flow1.destination, flow1.source, flow2.destination],
+                [flow1.source, flow2.destination, flow2.source, flow1.destination],
+                [flow2.destination, flow1.destination, flow2.source, flow1.source],
+                [flow1.source, flow2.source, flow1.destination, flow2.destination]
+        ]
+    }
+
+    void verifyEndpoints(response, flow1SrcExpected, flow1DstExpected, flow2SrcExpected, flow2DstExpected) {
+        assert response.firstFlow.source == flow1SrcExpected
+        assert response.firstFlow.destination == flow1DstExpected
+        assert response.secondFlow.source == flow2SrcExpected
+        assert response.secondFlow.destination == flow2DstExpected
+    }
+
+    void verifyEndpoints(flow1, flow2, flow1SrcExpected, flow1DstExpected, flow2SrcExpected, flow2DstExpected) {
+        def flow1Updated = northbound.getFlow(flow1.id)
+        def flow2Updated = northbound.getFlow(flow2.id)
+
+        assert flow1Updated.source == flow1SrcExpected
+        assert flow1Updated.destination == flow1DstExpected
+        assert flow2Updated.source == flow2SrcExpected
+        assert flow2Updated.destination == flow2DstExpected
+    }
+
+    void validateFlows(FlowCreatePayload flow1, FlowCreatePayload flow2) {
+        Wrappers.wait(RULES_DELETION_TIME + RULES_INSTALLATION_TIME) {
+            [flow1, flow2].each {
+                assert northbound.validateFlow(it.id).each { direction -> assert direction.asExpected }
+            }
+        }
+    }
+
+    void validateSwitches(SwitchPair switchPair) {
+        Wrappers.wait(RULES_DELETION_TIME + RULES_INSTALLATION_TIME) {
+            [switchPair.src, switchPair.dst].each {
+                def validationResult = northbound.validateSwitch(it.dpId)
```

Comment:
You must ensure that OF1.2 switch is not picked as src/dst in any of the scenarios above, otherwise this is gonna fail on stage

Reply:
Done.